plugins/registry.js

```diff
@@ -500,17 +501,28 @@ export default class PluginRegistry {
         store.dispatch(registerPluginTranslationsSource(this.id, getTranslationsForLocale));
     }
 
-    // Register a admin console definitions override function (This is a
-    // low-level interface and can change in the future).
-    // - func - A function that recieve the admin console config
-    // definitions and return a new version of it, which is used for build the
-    // admin console.
-    // Each plugin can only register a function, if the plugin register
-    // multiple functions the last one will be used.
+    // Register a admin console definitions override function
+    // Note that this is a low-level interface primarily meant for internal use, and is not subject
+    // to semver guarantees. It may change in the future.
+    // Accepts the following:
+    // - func - A function that recieve the admin console config definitions and return a new
+    //          version of it, which is used for build the admin console.
+    // Each plugin can register at most one admin console plugin function, with newer registrations
+    // replacing older ones.
     registerAdminConsolePlugin(func) {
         store.dispatch(registerAdminConsolePlugin(this.id, func));
     }
 
+    // Register a custom React component to manage the plugin configuration for the given setting key.
+    // Accepts the following:
+    // - key - A key specified in the settings_schema.settings block of the plugin's manifest.
+    // - component - A react component to render in place of the default handling.
+    // - title - Optional title that, if available, will be displayed on the left column
+    // on the settings page and the registered component will be rendered on space available in the right column.
+    registerAdminConsoleCustomSetting(key, component, title) {
```

Comment:
Upon reflecting on https://github.com/mattermost/mattermost-plugin-demo/pull/72, I feel like the change in semantics for defining `title` might be too strong. It also seems weird to have to define something that's already defined in the manifest to get this behaviour. I wonder if we should split up the functions instead?

Reply:
I've updated to receive a flag to `showTitle` and will display the `display_name` defined in the manifest for that setting.